### PR TITLE
Update config proto for destination policy

### PR DESCRIPTION
**Pull request by kyessenov**
Needs a follow-up to validate destination policy name against the destination field value.
As-is sufficient to pass the tests with no functional change.

### CODE REVIEW

**Comment by kyessenov** [comment]
@sebastienvas we need to do `bazel clean` to cleanup genfiles. don't know why bazel doesn't do that automatically.

**Comment by kyessenov** [comment]
jenkins rebuild manager/presubmit

File: model/config.go [hunk 1]
```diff
@@ -234,15 +234,16 @@ func (i *IstioRegistry) PoliciesByNamespace(namespace string) []*proxyconfig.Des
 	return out
 }
 
-// DestinationPolicies lists all policies for a service version.
-// Policies are not inherited by tags inclusion. The policy tags must match the tags precisely.
-func (i *IstioRegistry) DestinationPolicies(destination string, tags Tags) []*proxyconfig.DestinationPolicy {
-	out := make([]*proxyconfig.DestinationPolicy, 0)
+// DestinationPolicy returns a policy for a service version.
```
Comment (rshriram) [inline model/config.go]: Can you make a note here about renaming this to Destination Rule?

**Comment by rshriram** [comment]
Feel free to merge if you think these need to be handled elsewhere.

File: model/config.go (hunk 1 above)
Comment (kyessenov) [inline model/config.go]: one thing at a time...

**Comment by kyessenov** [comment]
I found only single use of policy in this repo. The semantics is that there should be only one dest-policy object per destination. I'll have to deal with naming YAMLs appropriate in a next PR.
For example:
```
destination: a.default.svc.cluster.local
policy:...
```
should a dest-policy named "a" in "default" namespace.

**Comment by kyessenov** [comment]
jenkins rebuild manager/presubmit

**Pull request merged**